### DISCUSSION TITLE
what's the difference between the tableColumns and partitionColumns ?

### DISCUSSION THREAD

**Discussion by jiangjiguang** [trinodb/trino#18063] [Q&A] [answered]
In my hive table, the tableColumns and partitionColumns are some,   
In what circumstances are they different ? 
Why have TableToPartitionMapping class ? 
@kokosing Can you help me ?

**Comment by kokosing** [comment, marked as answer]
Hive allows to modify the table and partition schema independently and it is possible that they go out of sync. Also hive allows to register new partitions that can also have different schema.

**Reply by jiangjiguang** [reply]
@kokosing  thanks, I got it
